webpack/assets/javascripts/react_app/components/hosts/powerStatus/index.js

```diff
@@ -1,43 +1,42 @@
-import { connect } from 'react-redux';
-import React from 'react';
+import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
+import { useSelector, useDispatch } from 'react-redux';
+import PowerStatus from './PowerStatus';
+import { getHostPowerState } from '../../../redux/actions/hosts/powerStatus/';
+import { HOST_POWER_STATUS } from './PowerStatusConstants';
+import {
+  selectError,
+  selectState,
+  selectTitle,
+  selectStatusText,
+} from './PowerStatusSelectors';
+
+const ConnectedPowerStatus = ({ data: { id, url } }) => {
+  const key = `${HOST_POWER_STATUS}_${id}`;
+  const error = useSelector(store => selectError(store, key));
+  const state = useSelector(store => selectState(store, key));
+  const title = useSelector(store => selectTitle(store, key));
+  const statusText = useSelector(store => selectStatusText(store, id));
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getHostPowerState(url, key));
+  }, [url, key, dispatch]);
```

Comment:
`dispatch` is not dependant by props or state, does it make sense to add it here?

Reply:
funny, I remember eslint was angry about that, or it was added by `lint -- --fix`

Reply:
we have an eslint rule that ask for it:
```
React Hook useEffect has a missing dependency: 'dispatch'. Either include it or remove the dependency array.eslint(react-hooks/exhaustive-deps)
```

Reply:
but dispatch would be different on every render (reference comparison) so the given `[]` is kinda useless...

Reply:
`useDispatch` returns a reference to the store, which doesn't change, so it won't be changed between rendering